scipy/stats/mstats_basic.py

```diff
@@ -543,21 +543,40 @@ def _spearmanr_2cols(x):
         return SpearmanrResult(rs, prob)
 
 
-def _kendall_p_exact(n, c):
-    # Exact p-value, see Maurice G. Kendall, "Rank Correlation Methods" (4th Edition), Charles Griffin & Co., 1970.
+def _kendall_p_exact(n, c, alternative='two-sided'):
+
+    # Use the fact that distribution is symmetric: always calculate a CDF in
+    # the left tail.
+    # This will be the one-sided p-value if `c` is on the side of
+    # the null distribution predicted by the alternative hypothesis.
+    # The two-sided p-value will be twice this value.
+    # If `c` is on the other side of the null distribution, we'll need to
+    # take the complement and add back the probability mass at `c`.
+    in_right_tail = (c >= (n*(n-1))//2 - c)
+    alternative_greater = (alternative == 'greater')
+    c = int(min(c, (n*(n-1))//2 - c))
+
+    # Exact p-value, see Maurice G. Kendall, "Rank Correlation Methods"
+    # (4th Edition), Charles Griffin & Co., 1970.
     if n <= 0:
         raise ValueError(f'n ({n}) must be positive')
     elif c < 0 or 4*c > n*(n-1):
         raise ValueError(f'c ({c}) must satisfy 0 <= 4c <= n(n-1) = {n*(n-1)}.')
     elif n == 1:
         prob = 1.0
+        p_mass_at_c = 1
```

Comment:
Would this case ever be reached? These lines haven't been covered by tests:

![image](https://user-images.githubusercontent.com/43181252/121536668-e5e9c880-ca20-11eb-854c-371311c1bb90.png)

From the code, it looks like `xtie`, `ytie`, and `tot` all will be 0 and the `if` condition will just return `nan`s. It probably doesn't need to be changed here thought. Just thought I would point this out.

Reply:
Yeah, I considered whether I should remove the case or add `p_mass_at_c` for it. I went with the latter as it was only one line and seemed slightly nicer to have a complete function (without a missing special case) even if won't be called with that input.

Reply:
> I went with the latter as it was very simple to add that one line and seemed slightly nicer to have a complete function (without a missing special case) even if won't be called with that input.

Makes sense. Thanks for clarifying!